src/components/Editor/Runners/HtmlRunner/HtmlRunner.jsx

```diff
@@ -104,21 +113,41 @@ function HtmlRunner() {
       if (typeof event.data?.msg === "string") {
         if (event.data?.msg === "ERROR: External link") {
           setError("externalLink");
-        } else {
+        } else if (event.data?.msg !== "Allowed external link") {
           setPreviewFile(`${event.data.payload.linkTo}.html`);
           dispatch(triggerCodeRun());
+        } else {
+          setExternalLink(event.data.payload.linkTo);
+          dispatch(triggerCodeRun());
         }
       }
     });
   };
 
   const iframeReload = () => {
     const iframe = output.current.contentDocument;
-    const filename = iframe.querySelectorAll("meta[filename]")[0]
-      ? iframe.querySelectorAll("meta[filename]")[0].getAttribute("filename")
-      : null;
-    if (runningFile !== filename) {
-      setRunningFile(filename);
+    if (!externalLink) {
+      const filename = iframe.querySelectorAll("meta[filename]")[0]
+        ? iframe.querySelectorAll("meta[filename]")[0].getAttribute("filename")
+        : null;
+      if (runningFile !== filename) {
+        setRunningFile(filename);
+      }
+    }
+    if (iframe) {
+      const linkElement = iframe.querySelector("a");
+
+      if (linkElement) {
+        linkElement.addEventListener("click", (e) => {
+          e.preventDefault();
+          const isExternalLink = matchingRegexes(
+            allowedExternalHrefs,
+            linkElement.href,
+          );
+          const target = isExternalLink ? "_blank" : "_top";
+          window.open(linkElement.getAttribute("href"), target);
+        });
```

Comment:
This needs to change in light of Mark's comments

Reply:
👍